### ISSUE TITLE
[Question | Gathering data] Are there use cases for change of cloud name?  

### ISSUE THREAD

**Issue by pdk27** [jenkinsci/ec2-fleet-plugin#382]
Context: https://github.com/jenkinsci/ec2-fleet-plugin/pull/383

Cloud name for EC2 fleet cloud is a special configuration. It is used to track the agents belonging to a particular cloud. Jenkins Cloud represents user configuration and is bound to the corresponding user data. i.e. a user-initiated change to the cloud configuration recreates that cloud's instance. Hence, cloud name is used to tag the cloud's agents to it. 

![image](https://github.com/jenkinsci/ec2-fleet-plugin/assets/3614196/1d52cd08-fb39-43c9-bcb3-d1cd60f505a6)

```
clouds:
  - eC2Fleet:
      name: "Jenkins-cloud-name-given-by-user-or-default-FleetCloud"
```

Current mitigation in place: 
- UI: duplicate cloud names are reported as an error and defaults suggested are unique  PRs: #392 
- Config as code: No mitigation yet. Issue: https://github.com/jenkinsci/ec2-fleet-plugin/issues/393 

Is modifying the cloud's name a common use case? Share your use case with us.

**Comment by wosiu** [comment]
are we talking here about fleet name? I mean this property in the config:
```
  clouds:
    - eC2Fleet:
        name: "some-name"
        fleet: "is-it-this-one-that-we-are-talking-about"
```
?

If so, I think it is acceptable to have some complicated procedure to change this name, as long as it's documented somewhere. Even if it requires some jenkins restart or some manual cleanup somewhere, and not just CasC reload. 
Because in our case changing this name does not happen often.

Probably cleaner solution though (if possible..) would be to have a separate property in the config, e.g.  "tracking-uuid" which would be documented to be immutable, with some example like random meaningless string. And for example put it into tags to match clouds that way. 
Because names might have some meaning which might require some changes after some time. While I don't see a need to change random string, even if the purpose of the given fleet changes over time. But that's a minor suggestion.

**Comment by wosiu** [comment]
EDIT, after giving a second though to my proposal above, I think it's an overkill and not ux friendly to have one more property for tracking, while still the name is needed. So I guess having a complicated procedure for renaming is good enough.

**Comment by pdk27** [comment]
@wosiu 
By cloud name, I mean this property in cloud configuration (NOT the fleet name ): 
![image](https://github.com/jenkinsci/ec2-fleet-plugin/assets/3614196/1d52cd08-fb39-43c9-bcb3-d1cd60f505a6)

```
clouds:
  - eC2Fleet:
      name: "Jenkins-cloud-name-given-by-user-or-default-FleetCloud"
```

I updated the issue description.
----

I agree with your point about another property not being a great choice for tracking. 
Using the`cloud name` to track the agents (i.e. EC2 instances) is better than using another config/ field like before `oldId`. Here is some more context around this recently merged change, if you are interested - #383.

**Comment by pdk27** [comment]
Closing this question as we have gathered some data.
